mautrix_facebook/portal.py

```diff
@@ -451,8 +463,9 @@ def optional_send_lock(self, user_id: str) -> Union[asyncio.Lock, FakeLock]:
     async def handle_matrix_leave(self, user: 'u.User') -> None:
         if self.is_direct:
             self.log.info(f"{user.mxid} left private chat portal with {self.fbid},"
-                          " cleaning up and deleting...")
-            await self.cleanup_and_delete()
+                          " cleaning up and deleting (if only puppets remain)...")
```

Comment:
~~The log should be inside the if/else and tell what's actually happening rather than what might happen depending on the room state.~~

This part is for direct chat portals (`if self.is_direct`). I'm pretty sure they should be cleaned up even if there are other users left. The part for group chat portals already doesn't clean up automatically.

Reply:
This was needed to make it safe to invite other users (including bots) to a direct chat portal. Otherwise, if any member of the room leaves or if someone rejects an invite to the room, the room gets shut down with no way of restoring it (since the cleanup process empties the room).

The purpose of adding extra Matrix users to a "direct" chat portal, as opposed to simply making a group portal, is to do things like add Matrix-only bots or invite the Matrix account of the person you're talking to on Facebook, all while allowing the room on Facebook's end to remain as a direct chat.

Reply:
That makes sense, but it should simply check if the user who left the room is the one who's actually participating in the facebook chat. If that user leaves, the portal should be deleted. If anyone leaves, it shouldn't do anything.

Reply:
Good idea. I'll get to it.

Reply:
58e691a